GPSlog_Basic/app/src/main/java/com/example/gpslog/SVMClassifier.java

```diff
@@ -49,25 +49,26 @@ public void classifyStopAndGo(){
     // where the Track's hidden state is either ACCELERATION or STOPPED
     // Each Segment will have characteristic features that can be used by SVM
     private ArrayList<Segment> getSegments(){
+
         ArrayList<Track> tracks= (ArrayList<Track>)dbh.getAllTracks();
         ArrayList<Segment> segments = new ArrayList<>();
-        boolean isSegment = false;
         ArrayList<Track> segTracks = new ArrayList<>();
-        for(int i = 1; i < tracks.size(); i++){
-            if((tracks.get(i - 1).hiddenState == HMMClassifier.FREEFLOW &&
-                    tracks.get(i).hiddenState == HMMClassifier.ACCELERATION) ||i == 1){
-                isSegment = true;
-            }
-            else if((isSegment && tracks.get(i).hiddenState == HMMClassifier.FREEFLOW) ||
-                    i == tracks.size() -1 ){
+        boolean isSegment = false;
+
+        //Makes a decision based on whether a segment is active, and if the current track is FreeFlow
+        for(int i = 0; i <= tracks.size(); i++){
```

Comment:
If I am correct, this for loop will go out of bounds when i == tracks.size() and it will throw an OutOfBoundsException.